lexos/processors/prepare/scrubber.py

```diff
@@ -693,43 +642,29 @@ def keep_words(text: str, non_removal_string: str) -> str:
 
     punctuation = get_punctuation_string()
 
-    split_lines = non_removal_string.split("\n")
-    keep_list = []
-    for line in split_lines:
-        line = line.strip()
-        # Using re for multiple delimiter splitting
-        line = re.split('[., ]', line)  # maybe change '[., ]' for punctuation
-        keep_list.extend(line)
-
-    keep_list = [word for word in keep_list if word != '']
+    # a list containing the words in non_removal_string without punctuation
+    # or whitespace
+    keep_list = split_input_word_string(input_string=non_removal_string)
 
     split_lines = text.split("\n")
-
-    text_list = []  # list containing all words in text
+    text_list = []  # list of words and ""
     for line in split_lines:
```

Comment:
This part looks a lot like `split_input_word_string`, can we abstract those two function into one function?